modules/observers/net_observer_reporter_print.cc

```diff
@@ -10,11 +11,44 @@ const std::string NetObserverReporterPrint::IDENTIFIER = "Caffe2Observer ";
 void NetObserverReporterPrint::report(
     NetBase* net,
     std::map<std::string, PerformanceInformation>& info) {
-  LOG(INFO) << IDENTIFIER << "Net Name - " << net->Name();
-  LOG(INFO) << IDENTIFIER << "Delay Start";
+  std::map<std::string, std::map<std::string, std::string>> caffe2_perf;
+  std::map<std::string, std::string> op_perf;
+  std::map<std::string, std::string> net_perf;
+
   for (auto& p : info) {
-    LOG(INFO) << IDENTIFIER << p.first << " - " << p.second.latency << "\t(ms)";
+    if ((p.first == "NET_DELAY") && (info.size() == 1)) {
```

Comment:
It's more clear if we handle `info.size()==1` outside the loop because there is no need to check it every iteration.

Reply:
Actually, 99% case, it will be short-circuit.